src/core/Akka/Util/Internal/TaskExtensions.cs

```diff
@@ -48,5 +49,26 @@ public static class TaskExtensions
                 }, TaskContinuationOptions.ExecuteSynchronously);
             return tcs.Task;
         }
+
+        /// <summary>
+        /// Returns the task which completes with result of original task if cancellation token not canceled it before completion.
+        /// </summary>
+        /// <param name="task">The original task.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The task which completes with result of original task or with cancelled state.</returns>
+        public static Task WithCancellation(this Task task, CancellationToken cancellationToken)
```

Comment:
@kostrse actually it's probably fine to leave this `public`. `WithCancellation` sounds great 👍